app/main.py

```diff
@@ -1 +1,27 @@
-# write your code here
+from __future__ import annotations
+from app.cafe import Cafe
+from app.errors import (
+    VaccineError,
+    NotWearingMaskError
+)
+
+
+def go_to_cafe(friends: list[dict], cafe: Cafe) -> str | VaccineError:
+    for friend in friends:
+        try:
+            cafe.visit_cafe(friend)
+        except VaccineError:
+            return "All friends should be vaccinated"
+        except NotWearingMaskError:
+            continue
+
+    for friend in friends:
+        try:
+            cafe.visit_cafe(friend)
+        except NotWearingMaskError:
+            unmasked = 0
+            for some_friend in friends:
+                if some_friend["wearing_a_mask"] is False:
+                    unmasked += 1
+            return f"Friends should buy {unmasked} masks"
+    return f"Friends can go to {cafe.name}"
```

Comment:
why did you use two loops and try/except statements?

Reply:
Because i have to move over all the friends in income dict. The first loop catch vaccine error, the second masks error, so because of order i've done two loops. Is it posible to avoid loops or all the loops?

Reply:
yep) write in the chat in slack)